src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ClusterUsersReconciler.java

```diff
@@ -0,0 +1,110 @@
+package com.indeed.operators.rabbitmq.reconciliation.rabbitmq;
+
+import com.google.common.collect.ImmutableSet;
+import com.indeed.operators.rabbitmq.api.RabbitMQApiClient;
+import com.indeed.operators.rabbitmq.api.RabbitMQPasswordConverter;
+import com.indeed.operators.rabbitmq.controller.SecretsController;
+import com.indeed.operators.rabbitmq.model.crd.rabbitmq.VhostPermissions;
+import com.indeed.operators.rabbitmq.model.rabbitmq.RabbitMQCluster;
+import com.indeed.operators.rabbitmq.model.rabbitmq.RabbitMQConnectionInfo;
+import com.indeed.operators.rabbitmq.model.rabbitmq.RabbitMQUser;
+import com.indeed.operators.rabbitmq.model.rabbitmq.api.User;
+import com.indeed.operators.rabbitmq.resources.RabbitMQSecrets;
+import com.indeed.operators.rabbitmq.resources.RabbitMQServices;
+import io.fabric8.kubernetes.api.model.Secret;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ClusterUsersReconciler {
+    private static final Set<String> READ_ONLY_USERS = ImmutableSet.of("rabbit", "monitoring");
+    private static final Logger log = LoggerFactory.getLogger(ClusterUsersReconciler.class);
+
+    private final SecretsController secretsController;
+    private final RabbitMQApiClient apiClient;
+    private final RabbitMQPasswordConverter passwordConverter;
+
+    public ClusterUsersReconciler(
+            final SecretsController secretsController,
+            final RabbitMQApiClient apiClient,
+            final RabbitMQPasswordConverter passwordConverter
+    ) {
+        this.secretsController = secretsController;
+        this.apiClient = apiClient;
+        this.passwordConverter = passwordConverter;
+    }
+
+    public void reconcile(final RabbitMQCluster cluster) {
+        final RabbitMQConnectionInfo connectionInfo = new RabbitMQConnectionInfo(
+                cluster.getName(),
+                cluster.getNamespace(),
+                RabbitMQServices.getDiscoveryServiceName(cluster.getName())
+        );
+
+        final Map<String, RabbitMQUser> expectedUsers = cluster.getUsers().stream().collect(Collectors.toMap(RabbitMQUser::getUsername, user -> user));
```

Comment:
yes, i realize this whole method is pretty disgusting. i'll refactor it in the future.